ToolBox/CoinApi/CoinApiDataQueueHandler.cs

```diff
@@ -0,0 +1,360 @@
+        private void SubscribeSymbols(List<Symbol> symbolsToSubscribe)
+        {
+            if (symbolsToSubscribe.Count == 0) return;
+
+            Log.Trace($"CoinApiDataQueueHandler.SubscribeSymbols(): {string.Join(",", symbolsToSubscribe)}");
+
+            var message = JsonConvert.SerializeObject(new HelloMessage
+            {
+                ApiKey = _apiKey,
+                Heartbeat = true,
+                SubscribeDataType = new[] { "trade", "quote" },
+                SubscribeFilterSymbolId = _subscribedSymbols.Select(_symbolMapper.GetBrokerageSymbol).ToArray()
+            });
+
+            _webSocket.Send(message);
+
+            _connectionHandler.EnableMonitoring(true);
+        }
+
+        private void OnMessage(object sender, WebSocketMessage e)
+        {
+            //Log.Trace(e.Message);
+
+            try
+            {
+                var jObject = JObject.Parse(e.Message);
+
+                var type = jObject["type"].ToString();
+                switch (type)
+                {
+                    case "trade":
+                        {
+                            var trade = jObject.ToObject<TradeMessage>();
+
+                            lock (_locker)
+                            {
+                                _ticks.Add(new Tick
+                                {
+                                    Symbol = _symbolMapper.GetLeanSymbol(trade.SymbolId, SecurityType.Crypto, string.Empty),
+                                    Time = trade.TimeExchange,
+                                    Value = trade.Price,
+                                    Quantity = trade.Size,
+                                    TickType = TickType.Trade
+                                });
+                            }
+
+                            _connectionHandler.KeepAlive();
+                            break;
+                        }
+
+                    case "quote":
+                        {
+                            var quote = jObject.ToObject<QuoteMessage>();
+
+                            lock (_locker)
+                            {
+                                _ticks.Add(new Tick
+                                {
+                                    Symbol = _symbolMapper.GetLeanSymbol(quote.SymbolId, SecurityType.Crypto, string.Empty),
+                                    Time = quote.TimeExchange,
+                                    AskPrice = quote.AskPrice,
+                                    AskSize = quote.AskSize,
+                                    BidPrice = quote.BidPrice,
+                                    BidSize = quote.BidSize,
+                                    TickType = TickType.Quote
+                                });
+                            }
+
+                            _connectionHandler.KeepAlive();
+                            break;
+                        }
+
+                    // not a typo :)
+                    case "hearbeat":
```

Comment:
haha, really! nice comment, but that's super strange that their api sends it back as `hearbeat` -- maybe we can add `heartbeat` as well just in case they decide to fix it:

``` csharp
case "hearbeat":
case "heartbeat":
```